doc/man7/SM2.pod

```diff
@@ -0,0 +1,77 @@
+=pod
+
+=head1 NAME
+
+SM2 - Chinese SM2 signature and encryption algorithm support
+
+=head1 DESCRIPTION
+
+B<SM2> algorithm is first defined by the Chinese national standard GM/T 0003-2012
+and is standardized by ISO as ISO/IEC 14888. B<SM2> is actually an elliptic curve
+based algorithm. The current implementation in OpenSSL supports both signature and
+encryption schemes via EVP interface.
+
+When doing the B<SM2> signature algorithm, it requires a distinguishing identifier
+to form the message prefix which is hashed before the real message is hashed.
+
+=head1 NOTES
+
+B<SM2> signature can be generated by using the 'DigestSign' series APIs, for instance,
+EVP_DigestSignInit(), EVP_DigestSignUpdate() and EVP_DigestSignFinal(). Ditto
+for the verification process by calling the 'DigestVerify' series APIs.
+
+There are several special steps need to be done before computing an B<SM2> signature.
+
+The B<EVP_PKEY> structure should be set to B<EVP_PKEY_SM2> by calling:
```

Comment:
The B\<EVP_PKEY\> structure will default to using ECDSA for signatures when it is created. It should be set...